dascore/core/spool.py

```
@@ -665,13 +667,101 @@ def get_contents(self) -> pd.DataFrame:
+        # select kwargs) can serve integer requests from the patch list.
+        patches = self._patches
+        if (
+            patches is not None
+            and not self._select_kwargs
+            and isinstance(item, int | np.integer)
+        ):
+            try:
+                return patches[item]
+            except IndexError:
+                msg = f"index of [{item}] is out of bounds for spool."
+                raise IndexError(msg)
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)